feat: full support for persistent HTML5 notifications (#35810)
electron/electron: docs/api/app.md

**Note:** Extra command line arguments might be added by Chromium,
such as `--original-process-start-time`.

> ### Event: 'notification-activation' _Windows_

Reviewer: All the parameters to this event should be on a single `details` object, not each as an individual parameter

Author: I've changed event prototype and name of parameters https://github.com/electron/electron/commit/de1f5b59c626d2be2213c25d33ce105ddcdfa5e6

Returns:

* `event` Event

> * `appUserModelId` string - contains App module ID

Author: I'm thinking about exclusion of these event at all, looks like we do not want it(event) and all my previous explanations were not taken into according.

> * `appUserModelId` string - contains App module ID

Author: This parameter allows us to distinguish notifications triggered from different apps, what if somebody will decide to create global notification manager interface and look on history? Have simple question : how we will find our notifications? I told this before - we need to deliver as much possible data for user as we can, we should not bound him into our "this minute" decisions. 
After few hours of thinking I can answer that : this parameter will stay into event, I will try to find better name for it.

> * `appUserModelId` string - contains App module ID

Author: I've changed event prototype and name of parameters de1f5b59c626d2be2213c25d33ce105ddcdfa5e6

> * `invokedArgs` string - contains arguments(in case of any) for toast button or data property in case of non-persistent notification

Reviewer: ```suggestion
* `activationArguments` string - contains arguments(in case of any) for toast button or data property in case of non-persistent notification
```

Author: event will be removed

Author: I've changed event prototype and name of parameters https://github.com/electron/electron/commit/de1f5b59c626d2be2213c25d33ce105ddcdfa5e6

> * `dataCount` integer - contains amount of token pairs present inputData string (not null, default value 0 - which means abcense of inputData)

Author: event will be removed

> * `dataCount` integer - contains amount of token pairs present inputData string (not null, default value 0 - which means abcense of inputData)

Author: dataCount - tells us real number of data pairs into inputData, this is necessary for exclusion problems related to JSON parsers, if into few words: "I do not want to rely on amount from some JSON parser I want to be sure that I've received everything that was packed".

> * `dataCount` integer - contains amount of token pairs present inputData string (not null, default value 0 - which means abcense of inputData)

Author: I've changed event prototype and name of parameters https://github.com/electron/electron/commit/de1f5b59c626d2be2213c25d33ce105ddcdfa5e6

> * `inputData` string - valid JSON-string which contains result of User input into toast Reply fields. Empty string in case of absence of User input (i.e for case dataCount == 0)

Reviewer: ```suggestion
* `notificationData` string - valid JSON-string which contains result of User input into toast Reply fields. Empty string in case of absence of User input (i.e for case dataCount == 0)
```

Author: event will be removed

Author: I've changed event prototype and name of parameters https://github.com/electron/electron/commit/de1f5b59c626d2be2213c25d33ce105ddcdfa5e6

> * `inputData` string - valid JSON-string which contains result of User input into toast Reply fields. Empty string in case of absence of User input (i.e for case dataCount == 0)

Author: event will be removed

> * `inputData` string - valid JSON-string which contains result of User input into toast Reply fields. Empty string in case of absence of User input (i.e for case dataCount == 0)

Author: I've changed event prototype and name of parameters https://github.com/electron/electron/commit/de1f5b59c626d2be2213c25d33ce105ddcdfa5e6

### `app.isHidden()` _macOS_

> Returns `boolean` - `true` if the application???including all of its windows???is hidden (e.g. with `Command-H`), `false` otherwise.

Reviewer: All these charset formatting issues need to be reverted across all the docs that your editor changed

Author: I've reverted charset changes, see commit https://github.com/electron/electron/commit/88f173b1bf6a883f0de2c06c1eeb74942a641bb0

### `app.isHidden()` _macOS_

> Returns `boolean` - `true` if the application-including all of its windows???is hidden (e.g. with `Command-H`), `false` otherwise.

Reviewer 2: ```suggestion
Returns `boolean` - `true` if the application - including all of its windows - is hidden (e.g. with `Command-H`), `false` otherwise.
```

Author: Hello @felixrieseberg thank you for the remark. 

Honestly due to continuous work of community into notifications module my branch becomes conflicting, and I have to resolve conflicts, sometimes in parts which are not related to notifications at all, this is a result of one of such resolving. Maybe I missed this moment, will take a look when have enough time.

You can use this property to prompt users to download the arm64 version of
your application when they are running the x64 version under Rosetta
incorrectly.

> ### `app.notificationsComServerCLSID` _Windows_

Reviewer: Per last review comment:

> Can we provide a link for where how to generate a CLSID, something like https://www.guidgenerator.com/online-guid-generator.aspx with Uppercase + Brackets + Hyphens checked is a reasonable way to do it.

> Or just documenting it's an upper case uuid with curly braces

This documentation is still required

Author: https://learn.microsoft.com/en-us/windows/win32/api/rpcdce/nf-rpcdce-uuidfromstring

Author: See commit https://github.com/electron/electron/commit/88f173b1bf6a883f0de2c06c1eeb74942a641bb0

> A `string` which contains fully formed CLSID for notifications COM server.
> This is necessary only for apps with persistent notifications support.
> The current default value for this property is `empty` (not `nil`). It means that by the default no persistent notifications support is exist.
> 
> Changing for property value should be provided earlier than `ready` event triggered. Preferably changing for `app.notificationsComServerCLSID` is need to be provided into `will-finish-launching` event handler.

Reviewer: ```suggestion
A `string` which contains a fully formed CLSID used for the notifications COM server.
This is only necessary for apps that need persistent notifications support.
The current default value for this property is an empty string `''`. This means that by default persistent notifications are not supported on Windows.

This value should be set before the `ready` event. To ensure this you can set the `app.notificationsComServerCLSID` property inside a `will-finish-launching` event handler.
```

Author: See commit https://github.com/electron/electron/commit/88f173b1bf6a883f0de2c06c1eeb74942a641bb0
